### PR TITLE
Add an additional setting for extra options to the Flix compiler

### LINKED ISSUE #118: Add an additional setting for extra options to the Flix compiler
So we have two boxes:

- Extra arguments for the JVM
- Extra arguments for the Flix compiler

### PR DESCRIPTION
**Pull request by manoj2601**
close #118 
A new box added.

Screenshot from 2021-07-08 17-28-23

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -126,6 +126,11 @@
           "type": "string",
           "default": "",
           "description": "Additional JVM arguments separated by spaces"
+        },
+        "flix.extraOptionsToFlixCompiler": {
+          "type": "string",
+          "default": "",
+          "description": "Additional Flix Compiler options separated by spaces"
```
Comment (magnus-madsen) [inline package.json:133]: Compiler -> compiler

File: client/src/handlers/handlers.ts [hunk 2]
```diff
@@ -55,6 +55,12 @@ function newFlixTerminal() {
     return terminal
 }
 
+
+
```
Comment (magnus-madsen) [inline client/src/handlers/handlers.ts:59]: Add function comment.

File: client/src/handlers/handlers.ts [hunk 3]
```diff
@@ -55,6 +55,12 @@ function newFlixTerminal() {
...
+function extraFlixOptions() {
```
Comment (magnus-madsen) [inline client/src/handlers/handlers.ts:60]: Rename: `extraFlixOptions` -> `getExtraFlixArgs`

**Review by magnus-madsen — COMMENTED** [review]
I added some minor suggestions for refactorings!

File: server/src/engine/flix.ts [hunk 4]
```diff
@@ -38,7 +38,8 @@ export interface CompileOnChange {
 export interface UserConfiguration {
   compileOnSave: CompileOnSave,
   compileOnChange: CompileOnChange,
-  extraArgumentsToTheJVM: string
+  extraArgumentsToTheJVM: string,
+  extraOptionsToFlixCompiler: string
 }
```
[the remark this answers is not in the document]
Comment (manoj2601) [reply server/src/engine/flix.ts:43]: Changed. The dialog boxes also changed automatically.
Screenshot from 2021-07-09 10-38-08

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**